Lockdep: increase size of reserved lockdep buffers for longer tests

Signed-off-by: Erez Zadok <ezk@cs.sunysb.edu>

diff --git a/kernel/lockdep_internals.h b/kernel/lockdep_internals.h
index 699a2ac..01787b1 100644
--- a/kernel/lockdep_internals.h
+++ b/kernel/lockdep_internals.h
@@ -45,6 +45,17 @@ enum {
 #define LOCKF_USED_IN_IRQ_READ \
 		(LOCKF_USED_IN_HARDIRQ_READ | LOCKF_USED_IN_SOFTIRQ_READ)
 
+#if 1
+#define MAX_LOCKDEP_ENTRIES	(16*8192UL)
+#undef MAX_LOCKDEP_KEYS_BITS
+#define MAX_LOCKDEP_KEYS_BITS	14
+#undef MAX_LOCKDEP_KEYS
+#define MAX_LOCKDEP_KEYS	(1UL << MAX_LOCKDEP_KEYS_BITS)
+#define MAX_LOCKDEP_CHAINS_BITS	17
+#define MAX_LOCKDEP_CHAINS	(1UL << MAX_LOCKDEP_CHAINS_BITS)
+#define MAX_LOCKDEP_CHAIN_HLOCKS (MAX_LOCKDEP_CHAINS*5)
+#define MAX_STACK_TRACE_ENTRIES	(32*262144UL)
+#else
 /*
  * MAX_LOCKDEP_ENTRIES is the maximum number of lock dependencies
  * we track.
@@ -66,6 +77,7 @@ enum {
  * addresses. Protected by the hash_lock.
  */
 #define MAX_STACK_TRACE_ENTRIES	262144UL
+#endif
 
 extern struct list_head all_lock_classes;
 extern struct lock_chain lock_chains[];
